### PR TITLE
JSDK-2355 QuickStart example for Network Quality API

### PR DESCRIPTION
**Pull request by manjeshbhargav**
@makarandp0 

This PR implements a QuickStart example for the Network Quality API.

## UI Snapshot

Screen Shot 2019-06-05 at 6 04 58 PM

## TODO

- [x] Add options to configure NetworkQualityVerbosity levels
- [x] Display the NetworkQualityStats for each RemoteParticipant
- [x] Add screenshot of the final UI

### CODE REVIEW

File: examples/networkquality/src/helpers.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+ * Change the local and remote Network Quality verbosity levels after joining the Room.
+ * @param {Room} room - The Room you just joined
+ * @param {number} localVerbosity - Verbosity level of Network Quality reports
+ *   for the LocalParticipant [1 - 3]
+ * @param {number} remoteVerbosity - Verbosity level of Network Quality reports
+ *   for the RemoteParticipant(s) [0 - 3]
+ * @returns {void}
+ */
+function setNetworkQualityConfiguration(room, localVerbosity, remoteVerbosity) {
+  room.localParticipant.setNetworkQualityConfiguration({
```
Comment (makarandp0) [inline examples/networkquality/src/helpers.js]: not related to your change, but it feels odd that we have a api on `localParticipant` that let you update the verbosity of remote participant. Wonder why don't we have this method in `room` instead?

File: examples/networkquality/src/index.js [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+
+  // Connect to a random Room with no media. This Participant will
+  // display the media of the other Participants that will enter
+  // the Room and watch for Network Quality updates.
+  someRoom = await connectToRoomWithNetworkQuality(
+    creds.token,
+    parseInt(localVerbosity.value, 10),
+    parseInt(remoteVerbosity.value, 10));
+
+  showParticipant(someRoom.localParticipant, false);
```
[the remark this answers is not in the document]
Comment (manjeshbhargav) [reply examples/networkquality/src/index.js]: @makarandp0 I think `room.localParticipant` indicates that it is local, don't you think?

File: examples/networkquality/src/index.js [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+
+  // Convert the "Create Room" button to a "Leave Room" button.
+  setupRoomBtn.onclick = teardownRoom;
+  setupRoomBtn.value = 'Leave Room';
+
+  // create controls to connect few users
+  ['Alice', 'Bob', 'Charlie', 'Mak'].forEach(createUserControls);
+
+  someRoom.on('participantConnected', function(participant) {
+    showParticipant(participant, true);
```
[the remark this answers is not in the document]
Comment (manjeshbhargav) [reply examples/networkquality/src/index.js]: I think "participantConnected" indicates that it is remote, don't you think?

File: examples/networkquality/src/helpers.js (hunk 1 above)
Comment (manjeshbhargav) [reply examples/networkquality/src/helpers.js]: @makarandp0 We are setting verbosity of reports for the RemoteParticipants, which are still sent to the LocalParticipant. This is independent of the RemoteParticipants' own configuration.

File: examples/networkquality/src/index.js [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+  container.appendChild(btn);
+  return btn;
+}
+
+/**
+ *
+ * Creates controls for user to disconnect from the Room.
+ */
+async function createUserControls(userIdentity) {
+  const creds = await getRoomCredentials(userIdentity);
```
[the remark this answers is not in the document]
Comment (manjeshbhargav) [reply examples/networkquality/src/index.js]: @makarandp0 Can you create a ticket to convert quickstart code to es6?

File: examples/index.html [hunk 5]
```diff
@@ -137,6 +137,24 @@ <h4 class="card-header">
...
+        <div class="card-deck">
+          <div class="card" style="text-decoration: none">
+            <a href="/networkquality">
+              <h4 class="card-header">
+                Network Quality
+              </h4>
+            </a>
+            <div class="card-block">
+              <p class="card-text">
+                This app demonstrates the Network Quality API for Group or Small Group Rooms.
```
[the remark this answers is not in the document]
Comment (manjeshbhargav) [reply examples/index.html]: @makarandp0 I think adding a link to the API guide would be better here. I will also do the same thing for other APIs.

File: examples/networkquality/src/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (makarandp0) [reply examples/networkquality/src/index.js]: my comment was about the literal false value - it not clear what that false parameter means, w/o  looking at the definition of `showParticipant`

File: examples/networkquality/src/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (makarandp0) [reply examples/networkquality/src/index.js]: my comment was about the literal true value - it not clear what that false parameter means, w/o  looking at the definition of `showParticipant`

**Pull request merged**